src/stores/WalletStore.ts

```diff
@@ -95,6 +102,48 @@ class WalletStore extends PersistableStore {
     throw new Error('Unknown proof type')
   }
 
+  private async checkNetwork(provider: Web3Provider) {
+    const network = env.VITE_ETH_NETWORK
+    const userNetwork = (await provider.getNetwork()).name
+    if (userNetwork === network) return (this.needNetworkChange = false)
+
+    this.needNetworkChange = true
+    await this.requestChangeNetwork(provider, network)
+  }
+
+  private async requestChangeNetwork(provider: Web3Provider, network: string) {
+    const networkName = network.charAt(0).toUpperCase() + network.slice(1)
+    const currency = `${networkName}ETH`
+    const chainId = hexValue(env.VITE_CHAIN_ID)
+
+    const blockExplorerUrl =
+      network === 'mainnet'
+        ? 'https://etherscan.io/'
+        : `https://${network}.etherscan.io/`
+
+    try {
+      await provider.send('wallet_switchEthereumChain', [{ chainId }])
+      this.needNetworkChange = false
+    } catch (error) {
+      const code = serializeError(error).code
+      if (code !== 4902) return
```

Comment:
Don't we need to set `needNetworkChange` to `true` here?

Reply:
it's already set in `#L110`